deployer/hub.py

```diff
@@ -149,147 +170,176 @@ def auth_aws(self):
         Finally get those environment variables to the original values to prevent
         side-effects on existing local configuration.
         """
-        config = self.spec['aws']
-        key_path = config['key']
-        cluster_type = config['clusterType']
-        cluster_name = config['clusterName']
-        region = config['region']
+        config = self.spec["aws"]
+        key_path = config["key"]
+        cluster_type = config["clusterType"]
+        cluster_name = config["clusterName"]
+        region = config["region"]
 
-        if cluster_type == 'kops':
-            state_store = config['stateStore']
+        if cluster_type == "kops":
+            state_store = config["stateStore"]
 
         with tempfile.NamedTemporaryFile() as kubeconfig:
-            orig_kubeconfig = os.environ.get('KUBECONFIG', None)
-            orig_access_key_id = os.environ.get('AWS_ACCESS_KEY_ID', None)
-            orig_secret_access_key = os.environ.get('AWS_SECRET_ACCESS_KEY', None)
+            orig_kubeconfig = os.environ.get("KUBECONFIG", None)
+            orig_access_key_id = os.environ.get("AWS_ACCESS_KEY_ID", None)
+            orig_secret_access_key = os.environ.get("AWS_SECRET_ACCESS_KEY", None)
             try:
                 with decrypt_file(key_path) as decrypted_key_path:
 
                     decrypted_key_abspath = os.path.abspath(decrypted_key_path)
                     if not os.path.isfile(decrypted_key_abspath):
-                        raise FileNotFoundError(
-                            'The decrypted key file does not exist')
+                        raise FileNotFoundError("The decrypted key file does not exist")
                     with open(decrypted_key_abspath) as f:
                         creds = json.load(f)
 
-                    os.environ['AWS_ACCESS_KEY_ID'] = creds["AccessKey"]['AccessKeyId']
-                    os.environ['AWS_SECRET_ACCESS_KEY'] = creds["AccessKey"]['SecretAccessKey']
-
-                os.environ['KUBECONFIG'] = kubeconfig.name
+                    os.environ["AWS_ACCESS_KEY_ID"] = creds["AccessKey"]["AccessKeyId"]
+                    os.environ["AWS_SECRET_ACCESS_KEY"] = creds["AccessKey"][
+                        "SecretAccessKey"
```

Comment:
This one is weird... but I guess I need to let it go 😜

Reply:
Yikes, yeah, that was a ugly outcome.

Reply:
Yeah, it's not always perfect, but I guess this what we sacrifice for automation. At least you never have to _write_ it that way 😝